packages/rlc-common/src/metadata/buildReadmeFile.ts

```diff
@@ -357,6 +364,63 @@ export function buildReadmeFile(model: RLCModel) {
   };
 }
 
+/**
+ * Updates an existing README.md file by only updating the "Key links:" section.
+ */
+export function updateReadmeFile(
+  model: RLCModel,
+  existingReadmeFilePath: string
+): { path: string; content: string } | undefined {
+  // Read existing content
+  let existingContent: string;
+  try {
+    existingContent = readFileSync(existingReadmeFilePath, "utf8");
+  } catch (e) {
+    // If the file doesn't exist, we don't need to update it.
+    return;
+  }
+
+  // Generate Key links
+  const metadata = createMetadata(model) ?? {};
+  const newKeyLinksSection = hbs.compile(
+    model.options && isAzurePackage(model)
+      ? model.options.isModularLibrary
+        ? keyLinksModularTemplate
+        : keyLinksRLCTemplate
+      : nonBrandedkeyLinksTemplate,
+    { noEscape: true }
+  )(metadata);
+  if (!newKeyLinksSection?.trim()) {
+    // Nothing to update, return existing content
+    return { path: "README.md", content: existingContent };
+  }
+
+  // Update and return
+  return {
+    path: "README.md",
+    content: updateKeyLinksSection(existingContent, newKeyLinksSection.trim())
+  };
+}
+
+/**
+ * Updates only the Key links section in the existing README content.
+ */
+function updateKeyLinksSection(
+  existingContent: string,
+  newKeyLinksSection: string
+): string {
+  // Match "Key links:" followed by content until the next section (## or #) or end of file
+  const keyLinksPattern = /Key links:\s*\n[\s\S]*?(?=\n## |\n# |$)/;
```

Comment:
Is it necessary if we only update `API reference documentation` link? It seems currently we would override all links and this may not work for data-plane SDKs.

<img width="1561" height="276" alt="image" src="https://github.com/user-attachments/assets/1ae95d82-0933-47ad-a368-8c5434af3ced" />